highfidelity/hifi
# Improve UX for snapshot load failure

Note: I'm starting to get uncomfortable with the confusing state of `snapshot.js` and `snapshotReview.js`. I want the code to be more clear. Just look at the number of arguments to `addImage()`...

That being said, I'm not going to refactor this application right now. I wrote this code with early requirements in mind, and the requirements have changed over time, causing the code to get less clear as I added features without rearchitecture.

I'm writing this note to record, for myself and for others, that the code for the Snapshot application is no longer up to my own standards, and it'd be useful to refactor it in the future.

Test Plan:
Clear your settings. You can use desktop or HMD mode for this test plan.

1. Using the Snapshot app, choose a Snapshot location and take a "Still + GIF" snapshot.
2. When the GIF finishes processing, verify that both images appear in the Snapshot app, and that both images have the "Share" overlay appear on top of them.
3. Close the Snapshot app.
4. From your file explorer, delete the GIF snapshot from your local disk.
5. Open the Snapshot app.
6. Verify that the Still snapshot still appears and has the Share overlay appear on top of it.
7. Verify that the GIF snapshot no longer appears, and is instead replaced by a placeholder image. Also verify that the toggleable Share overlay appears on top of this placeholder image, and, when clicked, some text appears that reads "Could not find last snapshot taken.".
8. Take another set of Still + GIF snapshots.
9. When the GIF finishes processing, verify that both images appear in the Snapshot app, and that both images have the "Share" overlay appear on top of them.
